includes/integration.php

```diff
@@ -0,0 +1,537 @@
+                $_COOKIE[$k] = $v;
+                return true;
+            }
+            return $this->woo->session->set($k, $v);
+        }
+        
+        public function add_item_to_cookie($data)
+        {
+            $items = $this->get_items_in_cookie();
+            if (empty($items)) $items = [];
+            array_push($items, $data);
+            $encoded_items = json_encode($items);
+            $this->session_set($this->get_cookie_name(), $encoded_items);
+        }
+        
+        public function get_items_in_cookie()
+        {
+            $items = array();
+            $data = $this->session_get($this->get_cookie_name());
+            if (!empty($data)) {
+                $items = json_decode($data, true);
+            }
+            
+            return $items;
+        }
+        
+        public function get_identify_data_in_cookie()
+        {
+            $identify = [];
+            $data = $this->session_get($this->get_do_identify_cookie_name());
+            if (!empty($data)) {
+                $identify = json_decode($data, true);
+            }
+            return $identify;
+        }
+        
+        public function clear_items_in_cookie()
+        {
+            $this->session_set($this->get_cookie_name(), json_encode([]));
+            $this->session_set($this->get_do_identify_cookie_name(), json_encode([]));
+        }
+        
+        public function get_order_ip($order_id)
+        {
+            $ip_address = get_post_meta($order_id, '_customer_ip_address', true);
+            if (strpos($ip_address, '.') !== false) {
+                return $ip_address;
+            }
+            return false;
+        }
+        
+        public function process_cookie_events()
+        {
+            $items = $this->get_items_in_cookie();
+            if (count($items) > 0) {
+                $this->has_events_in_cookie = true;
+                foreach ($items as $event) {
+                    // put event in queue for sending to the JS library
+                    $this->put_event_in_queue($event);
+                }
+                
+            }
+            
+            // check if identify data resides in the session
+            $identify_data = $this->get_identify_data_in_cookie();
+            if(!empty($identify_data)) {
+                $this->identify_call_data = $identify_data;
+            }
+        }
+        
+        public function check_for_metrilo_clear()
+        {
+            if (!empty($_REQUEST) && !empty($_REQUEST['metrilo_clear'])) {
+                $this->clear_items_in_cookie();
+                wp_send_json_success();
+            }
+        }
+        
+        public function setup_admin_pages()
+        {
+            add_submenu_page('tools.php', 'Export to Metrilo', 'Export to Metrilo', 'export', 'metrilo-import', array($this, 'metrilo_import_page'));
+        }
+        
+        public function metrilo_import_page()
+        {
+            wp_enqueue_script('jquery');
```

Comment:
jQuery?

Reply:
Wordpress is managing some dependencies like jquery and we can use them for the import script in admin. In this case we ensure that jquery is present if admin import page is accessed.